### PR TITLE
api: support custom coeffs for div/grad/curl

### PR DESCRIPTION
**Pull request by mloubout**
See you tests for API

### CODE REVIEW

**Review by georgebisbas — APPROVED** [review]

File: devito/finite_differences/differentiable.py [hunk 1]
```diff
@@ -309,16 +309,22 @@ def laplacian(self, shift=None, order=None):
         order: int, optional, default=None
             Discretization order for the finite differences.
             Uses `func.space_order` when not specified
+        method: str, optional, default='FD'
+            Discretization method. Options are 'FD' (default) and
+            'RSFD' (rotated staggered grid finite-difference).
+        weights/w: list, tupe, or dict, optional, default=None
```
Comment (EdCaunt) [inline devito/finite_differences/differentiable.py:315]: Typo: "tupe" -> "tuple"

File: devito/finite_differences/differentiable.py [hunk 2]
```diff
@@ -358,13 +367,16 @@ def grad(self, shift=None, order=None, method='FD'):
         method: str, optional, default='FD'
             Discretization method. Options are 'FD' (default) and
             'RSFD' (rotated staggered grid finite-difference).
+        weights/w: list, tupe, or dict, optional, default=None
```
Comment (EdCaunt) [inline devito/finite_differences/differentiable.py:370]: Typo: "tupe" -> "tuple"

File: devito/finite_differences/operators.py [hunk 3]
```diff
@@ -14,9 +14,12 @@ def div(func, shift=None, order=None, method='FD'):
     method: str, optional, default='FD'
         Discretization method. Options are 'FD' (default) and
         'RSFD' (rotated staggered grid finite-difference).
+    weights/w: list, tupe, or dict, optional, default=None
```
Comment (EdCaunt) [inline devito/finite_differences/operators.py:17]: Ditto

File: devito/finite_differences/operators.py [hunk 4]
```diff
@@ -54,9 +57,12 @@ def grad(func, shift=None, order=None, method='FD'):
...
```
Comment (EdCaunt) [inline devito/finite_differences/operators.py:60]: Looks like the typo was copied into all the docstrings

Comment (mloubout) [reply devito/finite_differences/operators.py]: Fixed

**Review by EdCaunt — APPROVED** [review]

**Pull request merged**